torch/_jit_internal.py

```diff
@@ -160,11 +156,92 @@ def fn(*args, **kwargs):
     return fn
 
 
-def ignore(fn):
-    ignored_fns.add(fn)
+
+class FunctionModifiers(object):
+    """
+    Used to denote the behavior of a function in TorchScript. See export() and
+    ignore() for details.
+    """
+    IGNORE_AND_DROP = "ignore (leave as a call to Python, replace with a 'raise' on torch.jit.save)"
+    IGNORE = "ignore (leave as a call to Python, cannot be torch.jit.save'd)"
+    EXPORT = "export (compile this function even if nothing calls it)"
+    DEFAULT = "default (compile if called from a exported function / forward)"
+
+
+def export(fn):
+    """
+    This decorator indicates that a method is used as an entry point into a
+    ScriptModule. `forward` implicitly is used as an entry point, so it does
+    not need this decorator.
+
+    Methods are added to a ScriptModule as they are called in Python. If a
+    method is never called, it will not be included in the ScriptModule when
+    saving. This decorator explicitly marks that a method should be included
+    even if it is not called from Python.
+    """
+    fn._torchscript_modifier = FunctionModifiers.EXPORT
     return fn
 
 
+def ignore(maybe_fn=None, *, drop_on_export=False):
```

Comment:
kw only arg is not supported in python 2

Reply:
Unlanding now